### DISCUSSION TITLE
I need help - Validation error - BR-S-08 & BR-CO-10

### DISCUSSION THREAD

**Discussion by BeSt1979** [stephanstapel/ZUGFeRD-csharp#405] [Q&A]
Hello dear developpers, dear @stephanstapel  and community

My name is Benjamin. I'm having massive difficulties creating an "xRechnung" and I'm begging for help.

The resulting XML file is being rejected during validation due to errors BR-S-8 and BR-CO-10.

According to the description, the sum of the net line prices should not match the net total of the invoice. But I simply cannot find an error in the XML.

On the help pages of a software company I found the information that the invoice is expected as a net invoice, but a gross invoice was transferred. But I can't find anything I can do about this either.

I really need urgent help and beg someone to take a look at the XML file I have created and perhaps tell me what exactly is wrong here.

In hopeful expectation of an answer.

Regards,
Benjamin

`<?xml version="1.0" encoding="utf-8"?>
<rsm:CrossIndustryInvoice xmlns:a="urn:un:unece:uncefact:data:standard:QualifiedDataType:100" xmlns:rsm="urn:un:unece:uncefact:data:standard:CrossIndustryInvoice:100" xmlns:qdt="urn:un:unece:uncefact:data:standard:QualifiedDataType:100" xmlns:ram="urn:un:unece:uncefact:data:standard:ReusableAggregateBusinessInformationEntity:100" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:udt="urn:un:unece:uncefact:data:standard:UnqualifiedDataType:100">
	<rsm:ExchangedDocumentContext>
		<ram:BusinessProcessSpecifiedDocumentContextParameter>
			<ram:ID>urn:fdc:peppol.eu:2017:poacc:billing:01:1.0</ram:ID>
		</ram:BusinessProcessSpecifiedDocumentContextParameter>
		<ram:GuidelineSpecifiedDocumentContextParameter>
			<ram:ID>urn:cen.eu:en16931:2017#compliant#urn:xeinkauf.de:kosit:xrechnung_3.0</ram:ID>
		</ram:GuidelineSpecifiedDocumentContextParameter>
	</rsm:ExchangedDocumentContext>
	<rsm:ExchangedDocument>
		<ram:ID>RE 123456</ram:ID>
		<ram:TypeCode>380</ram:TypeCode>
		<ram:IssueDateTime>
			<udt:DateTimeString format="102">20240930</udt:DateTimeString>
		</ram:IssueDateTime>
	</rsm:ExchangedDocument>
	<rsm:SupplyChainTradeTransaction>
		<ram:IncludedSupplyChainTradeLineItem>
			<ram:AssociatedDocumentLineDocument>
				<ram:LineID>1</ram:LineID>
			</ram:AssociatedDocumentLineDocument>
			<ram:SpecifiedTradeProduct>
				<ram:Name>Sammelposition</ram:Name>
			</ram:SpecifiedTradeProduct>
			<ram:SpecifiedLineTradeAgreement>
				<ram:NetPriceProductTradePrice>
					<ram:ChargeAmount>2605.50</ram:ChargeAmount>
					<ram:BasisQuantity unitCode="NAR">1.0000</ram:BasisQuantity>
				</ram:NetPriceProductTradePrice>
			</ram:SpecifiedLineTradeAgreement>
			<ram:SpecifiedLineTradeDelivery>
				<ram:BilledQuantity unitCode="NAR">1.0000</ram:BilledQuantity>
			</ram:SpecifiedLineTradeDelivery>
			<ram:SpecifiedLineTradeSettlement>
				<ram:ApplicableTradeTax>
					<ram:TypeCode>VAT</ram:TypeCode>
					<ram:CategoryCode>S</ram:CategoryCode>
					<ram:RateApplicablePercent>19.00</ram:RateApplicablePercent>
				</ram:ApplicableTradeTax>
				<ram:SpecifiedTradeSettlementLineMonetarySummation>
					<ram:LineTotalAmount>3100.55</ram:LineTotalAmount>
				</ram:SpecifiedTradeSettlementLineMonetarySummation>
			</ram:SpecifiedLineTradeSettlement>
		</ram:IncludedSupplyChainTradeLineItem>
		<ram:ApplicableHeaderTradeAgreement>
			<ram:BuyerReference>991-15601-05</ram:BuyerReference>
			<ram:SellerTradeParty>
				<ram:ID>1234</ram:ID>
				<ram:Name>Firma GmbH</ram:Name>
				<ram:DefinedTradeContact>
					<ram:PersonName>Max Mustermann</ram:PersonName>
					<ram:TelephoneUniversalCommunication>
						<ram:CompleteNumber>0123/4567890</ram:CompleteNumber>
					</ram:TelephoneUniversalCommunication>
					<ram:EmailURIUniversalCommunication>
						<ram:URIID>mustermann@xyz.de</ram:URIID>
					</ram:EmailURIUniversalCommunication>
				</ram:DefinedTradeContact>
				<ram:PostalTradeAddress>
					<ram:PostcodeCode>70000</ram:PostcodeCode>
					<ram:LineOne>Teststrasse 1</ram:LineOne>
					<ram:CityName>Musterort</ram:CityName>
					<ram:CountryID>DE</ram:CountryID>
				</ram:PostalTradeAddress>
				<ram:SpecifiedTaxRegistration>
					<ram:ID schemeID="FC">123/456/7890</ram:ID>
				</ram:SpecifiedTaxRegistration>
				<ram:SpecifiedTaxRegistration>
					<ram:ID schemeID="VA">DE123456789</ram:ID>
				</ram:SpecifiedTaxRegistration>
			</ram:SellerTradeParty>
			<ram:BuyerTradeParty>
				<ram:ID>1234/555555</ram:ID>
				<ram:Name>Kundenname Testkunde</ram:Name>
				<ram:PostalTradeAddress>
					<ram:PostcodeCode>80000</ram:PostcodeCode>
					<ram:LineOne>Kundenstrasse 1</ram:LineOne>
					<ram:CityName>Kundenort</ram:CityName>
					<ram:CountryID>DE</ram:CountryID>
				</ram:PostalTradeAddress>
			</ram:BuyerTradeParty>
		</ram:ApplicableHeaderTradeAgreement>
		<ram:ApplicableHeaderTradeDelivery>
			<ram:ActualDeliverySupplyChainEvent>
				<ram:OccurrenceDateTime>
					<udt:DateTimeString format="102">20240930</udt:DateTimeString>
				</ram:OccurrenceDateTime>
			</ram:ActualDeliverySupplyChainEvent>
		</ram:ApplicableHeaderTradeDelivery>
		<ram:ApplicableHeaderTradeSettlement>
			<ram:PaymentReference>RE 123456 KndNr 1234/555555</ram:PaymentReference>
			<ram:InvoiceCurrencyCode>EUR</ram:InvoiceCurrencyCode>
			<ram:SpecifiedTradeSettlementPaymentMeans>
				<ram:TypeCode>58</ram:TypeCode>
				<ram:PayeePartyCreditorFinancialAccount>
					<ram:IBANID>DE02600501010002034304</ram:IBANID>
				</ram:PayeePartyCreditorFinancialAccount>
				<ram:PayeeSpecifiedCreditorFinancialInstitution>
					<ram:BICID>SOLADEST600</ram:BICID>
				</ram:PayeeSpecifiedCreditorFinancialInstitution>
			</ram:SpecifiedTradeSettlementPaymentMeans>
			<ram:ApplicableTradeTax>
				<ram:CalculatedAmount>495.05</ram:CalculatedAmount>
				<ram:TypeCode>VAT</ram:TypeCode>
				<ram:BasisAmount>2605.50</ram:BasisAmount>
				<ram:CategoryCode>S</ram:CategoryCode>
				<ram:RateApplicablePercent>19.00</ram:RateApplicablePercent>
			</ram:ApplicableTradeTax>
			<ram:SpecifiedTradePaymentTerms>
				<ram:Description>Zahlbar innerhalb 30 Tagen auf genanntes Konto</ram:Description>
				<ram:DueDateDateTime>
					<udt:DateTimeString format="102">20241030</udt:DateTimeString>
				</ram:DueDateDateTime>
			</ram:SpecifiedTradePaymentTerms>
			<ram:SpecifiedTradeSettlementHeaderMonetarySummation>
				<ram:LineTotalAmount>2605.50</ram:LineTotalAmount>
				<ram:ChargeTotalAmount>0.00</ram:ChargeTotalAmount>
				<ram:AllowanceTotalAmount>0.00</ram:AllowanceTotalAmount>
				<ram:TaxBasisTotalAmount>2605.50</ram:TaxBasisTotalAmount>
				<ram:TaxTotalAmount currencyID="EUR">495.05</ram:TaxTotalAmount>
				<ram:GrandTotalAmount>3100.55</ram:GrandTotalAmount>
				<ram:TotalPrepaidAmount>0.00</ram:TotalPrepaidAmount>
				<ram:DuePayableAmount>3100.55</ram:DuePayableAmount>
			</ram:SpecifiedTradeSettlementHeaderMonetarySummation>
		</ram:ApplicableHeaderTradeSettlement>
	</rsm:SupplyChainTradeTransaction>
</rsm:CrossIndustryInvoice>`

**Comment by IsThatUniqueEnough** [comment]
You receive both errors because you have included the tax amount in BT-131 "Invoice line _net_ amount" (`SpecifiedTradeSettlementLineMonetarySummation/LineTotalAmount`).

To make the document valid you also need to provide electronic addresses for seller and buyer.

**Comment by BeSt1979** [comment]
Thank you for that quick response.
I already added buyer- und seller-electronic addresses

But would you mind helping me in finding out where the "error" occurs in my code?

I add a invoice line with
`desc.AddTradeLineItem(...)`

Then I set totals with
`desc.SetTotals(netVal, 0, 0, vatVal, VATval, grossVal, 0.00, grossVal)`

And then I add ApplicableTradeTax with
`desc.AddApplicableTradeTax(netVal, 19.0, TaxTypes.VAT, TaxCategoryCodes.S)`

Which of these lines of code causes the incorrect entries in the XML?

**Comment by IsThatUniqueEnough** [comment]
I don't use ZUGFeRD-csharp but the error is in the line item. You provide a gross amount for BT-131 while a net amount is expected.

**Comment by BerndRiemke** [comment]
Is the problem fixed?
I become the same error...
What is the right code for this?
